sycl/test/format.py

```diff
@@ -5,26 +5,57 @@
 import os
 import re
 
-SUFFIXES = {".hpp"}
-
-
 class SYCLHeadersTest(lit.formats.TestFormat):
     def getTestsForPath(
         self, testSuite, path_in_suite, filepath, litConfig, localConfig
     ):
+        # path_in_suite is a tuple like:
+        #   ('self-contained-headers', 'path/to', 'file.hpp')
+        test_path = testSuite.getSourcePath(path_in_suite) + ".cpp"
+        if os.path.exists(test_path):
+            # We have a dedicated special test for a header, let's use a file
+            # from the suite itself
+
+            # None is a special value we use to distinguish those two cases
+            filepath = None
+            # The actual file has .cpp extension as every other test
+            path_in_suite = path_in_suite[:-1] + (path_in_suite[-1] + ".cpp",)
+        else:
+            # We don't have a dedicated special test for a header, therefore we
+            # fallback to a generalized version of it
+
+            # SYCL headers may depend on some generated files and therefore we
+            # use headers from the build folder for testing
+            filepath = os.path.join(localConfig.sycl_include, *path_in_suite[1:])
+
         yield lit.Test.Test(testSuite, path_in_suite, localConfig, file_path=filepath)
 
     def getTestsInDirectory(self, testSuite, path_in_suite, litConfig, localConfig):
-        # We traverse build/sycl/include/sycl directory
-        source_path = os.path.join(localConfig.sycl_include, "sycl")
+        # To respect SYCL_LIB_DUMPS_ONLY mode
+        if ".cpp" not in localConfig.suffixes:
+            return
```

Comment:
This variable makes little sense to me... What's wrong with `LIT_FILTER=.dump` ?